frontend/src/components/posting/reply.js

```diff
@@ -96,6 +102,17 @@ export default class extends Form {
   }
 
   onCancel = () => {
+    const editorEmpty =
+      this.state.post.length === 0 && this.state.attachments.length === 0
+
+    // If only the quote text is on editor user didn't add anything
+    // so no changes to discard
+    const onlyQuoteTextInEditor = this.state.post === this.quoteText
+
+    if (editorEmpty || onlyQuoteTextInEditor) {
```

Comment:
Shouldn't this be `&&`? `this.state.attachments.length === 1` will cause `editorEmpty` to be false, but if both `post` and `quoteText` are empty strings, `onlyQuoteTextInEditor` will be `true`?

Reply:
I've updated the conditional to make it simpler. Now it checks if `onlyQuoteTextInEditor` and `this.state.attachments === 0`.